collect_app/src/main/java/org/odk/collect/android/formmanagement/FormNavigator.kt

```diff
@@ -1,71 +1,34 @@
 package org.odk.collect.android.formmanagement
 
+import android.app.Activity
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
-import org.odk.collect.android.activities.FormEntryActivity
-import org.odk.collect.android.external.FormsContract
+import org.odk.collect.android.external.FormUriActivity
 import org.odk.collect.android.external.InstancesContract
-import org.odk.collect.android.utilities.ApplicationConstants.BundleKeys.FORM_MODE
-import org.odk.collect.android.utilities.ApplicationConstants.FormModes.VIEW_SENT
-import org.odk.collect.forms.instances.Instance
-import org.odk.collect.forms.instances.InstancesRepository
-import org.odk.collect.settings.SettingsProvider
-import org.odk.collect.settings.keys.ProtectedProjectKeys.KEY_EDIT_SAVED
-
-class FormNavigator(
-    private val projectId: String,
-    private val settingsProvider: SettingsProvider,
-    private val instancesRepositoryProvider: () -> InstancesRepository
-) {
-
-    fun editInstance(context: Context, instanceId: Long) {
-        val editingDisabled = !settingsProvider.getProtectedSettings().getBoolean(KEY_EDIT_SAVED)
-        val status = instancesRepositoryProvider().get(instanceId)?.status
-
-        context.startActivity(
-            editInstanceIntent(context, projectId, instanceId, editingDisabled, status)
-        )
-    }
-
-    fun newInstance(context: Context, formId: Long) {
-        context.startActivity(
-            newInstanceIntent(context, projectId, formId)
-        )
-    }
-
-    companion object {
-        fun newInstanceIntent(context: Context, uri: Uri?): Intent {
-            return Intent(context, FormEntryActivity::class.java).also {
-                it.action = Intent.ACTION_EDIT
-                it.data = uri
-            }
-        }
-
-        fun newInstanceIntent(context: Context, projectId: String, formId: Long): Intent {
-            return newInstanceIntent(context, FormsContract.getUri(projectId, formId))
+import kotlin.reflect.KClass
+
+object FormNavigator {
```

Comment:
I don't know if I want to make `FormUriActivity` bigger. It will probably grow in the future either way if we start adding more checks to make starting forms safer. Plus that would also mean moving tests to `FormUriActivityTest` and mixing them with those that check if a form can be started.

Reply:
Yeah that seems fair. We can just keep it this way for the moment. Maybe because just returns `Intent` objects rather than doing any actual navigating we should rename it to `FormFillingIntentFactory` or something?

Reply:
Ok done.